cds_books/ldap/cli.py

```diff
@@ -81,51 +44,18 @@ def import_users():
 
     click.secho(
         "--- Finished in %s seconds ---" % (time.time() - start_time),
-        fg="green"
+        fg="green",
     )
 
 
 @ldap_users.command(name="sync")
 @with_appcontext
-def sync_users():
-    """Sync ldap with system users command."""
-    import time
-    start_time = time.time()
-
-    ldap_url = current_app.config["CDS_BOOKS_LDAP_URL"]
-    ldap_client = LdapClient(ldap_url)
-    system_users = RemoteAccount.query.join(User).all()
-    ldap_users = ldap_client.get_primary_accounts()
-
-    click.echo("Users fetched")
-
-    ldap_users_map = {}
-
-    for ldap_user in ldap_users:
-        ldap_person_id = ldap_user["employeeID"][0].decode("utf8")
-        ldap_users_map.update({ldap_person_id: ldap_user})
-
-    for system_user in system_users:
-        system_user_person_id = system_user.extra_data["person_id"]
-        ldap_user = ldap_users_map.get(system_user_person_id)
-        if ldap_user:
-            check_user_for_update(system_user, ldap_user)
-        else:
-            click.secho("Deleting user {} with ccid {}".format(
-                system_user.user, system_user_person_id), fg="red")
-            delete_user(system_user)
-
-    # Check if any ldap user is not in our system
-
-    for ldap_user in ldap_users:
-        ldap_mail = ldap_user["mail"][0].decode("utf8")
-        try:
-            User.query.filter(User.email == ldap_mail).one()
-        except NoResultFound:
-            click.secho("Adding new user {}".format(ldap_mail), fg="green")
-            import_ldap_users([ldap_user])
-
-    click.secho(
-        "--- Finished in %s seconds ---" % (time.time() - start_time),
-        fg="green"
-    )
+def ldap_sync_users():
+    """Add entry to database with info about this synchronization."""
+    log = LdapSynchronizationLog.create_cli()
+    result = sync_users()
+    print(result)
+    if result is None:
+        log.transition_fails()
```

Comment:
do we have any useful information that we can pass in the case that the task fails? Similar to the case it succeeds below?

Reply:
We could return the exception type & message as a string; but we agreed that exceptions would be rare and the message not necessarily useful to the end user. Moreover, the logger already records the stacktrace via Sentry.

Reply:
I added a new `message` field taking the value `${exception_type}: ${exception_message}` if an exception is caught or `NULL` otherwise. The full stacktrace is still available on the above-mentioned service.
![image](https://user-images.githubusercontent.com/9027075/90232925-a0b73300-de1d-11ea-875e-113e8739e9d4.png)